detox/src/devices/DeviceRegistry.js

```diff
@@ -120,16 +131,30 @@ class DeviceRegistry {
     return [];
   }
 
-  /***
+  /**
+   * @private
+   */
+  _registerDevice(deviceId, data) {
+    const state = this._lockfile.read();
+    let newState = _.reject(state, [FIELD_NAME_ID, deviceId]);
```

Comment:
Shouldn't we be signalizing if we are registering an already registered device?

Reply:
Probably so, but that's slightly out of scope, as it breaks the module's contract. Need to consider that separately.